src/main/java/org/folio/circulation/resources/RequestFromRepresentationService.java

```diff
@@ -150,51 +163,63 @@
   }
 
   private CompletableFuture<Result<Request>> fetchItemAndLoanForPageTlrRequest(Request request) {
-    if (request.getOperation() == Request.Operation.CREATE) {
-      return fromFutureResult(fetchItemForPageTlr(request)
-        .thenApply(r -> r.mapFailure(err -> errorHandler.handleValidationError(err,
-          NO_AVAILABLE_ITEMS_FOR_TLR, r))))
-        .flatMapFuture(this::fetchFirstLoanForUserWithTheSameInstanceId)
-        .flatMapFuture(this::fetchUserForLoan)
-        .toCompletableFuture();
-    } else {
-      return fromFutureResult(fetchItemForRequest(request))
-        .flatMapFuture(req -> succeeded(req).after(loanRepository::findOpenLoanForRequest)
-          .thenApply(r -> r.map(req::withLoan)))
-        .flatMapFuture(this::fetchUserForLoan)
-        .toCompletableFuture();
-    }
+    return request.getOperation() == Request.Operation.CREATE
+      ? findItemForTlrCreateOperation(request)
+      : findItemForTlrReplaceOperation(request);
+  }
+
+  private CompletableFuture<Result<Request>> findItemForTlrCreateOperation(Request request) {
```

Comment:
Maybe "..ForPageTlr.." to be more specific?

Reply:
renamed